Build failure: osl

### Steps To Reproduce

```
nix build github:nixos/nixpkgs/21735d31cf4b091d69f5dd24d5a06a874eb33499#osl
```

### Build log

```
CMake Error at /nix/store/ljk2a5s49msk252is7zw3kwq5dxdbkdr-openimageio-2.4.15.0-dev/lib/cmake/OpenImageIO/OpenImageIOConfig.cmake:43:
  Parse error.  Function missing ending ")".  End of file reached.
Call Stack (most recent call first):
  src/cmake/checked_find_package.cmake:127 (find_package)
  src/cmake/externalpackages.cmake:107 (checked_find_package)
  CMakeLists.txt:172 (include)


CMake Error at src/cmake/checked_find_package.cmake:127 (find_package):
  find_package Error reading CMake code from
  "/nix/store/ljk2a5s49msk252is7zw3kwq5dxdbkdr-openimageio-2.4.15.0-dev/lib/cmake/OpenImageIO/OpenImageIOConfig.cmake".
Call Stack (most recent call first):
  src/cmake/externalpackages.cmake:107 (checked_find_package)
  CMakeLists.txt:172 (include)


-- OpenImageIO library not found 
--     Try setting OpenImageIO_ROOT ? 
--     Maybe this will help:  src/build-scripts/build_openimageio.bash 
CMake Error at src/cmake/checked_find_package.cmake:177 (message):
  OpenImageIO is required, aborting.
Call Stack (most recent call first):
  src/cmake/externalpackages.cmake:107 (checked_find_package)
  CMakeLists.txt:172 (include)
```

### Additional context

`/nix/store/ljk2a5s49msk252is7zw3kwq5dxdbkdr-openimageio-2.4.15.0-dev/lib/cmake/OpenImageIO/OpenImageIOConfig.cmake:43` rendered in `micro`:

![image](https://github.com/NixOS/nixpkgs/assets/140964/0f5db3f0-b7ce-4982-8377-7c39e1b075d0)

bumping osl to `1.13.4.0-dev` did not work.

Applying https://github.com/AcademySoftwareFoundation/OpenShadingLanguage/pull/1742 did not work


### Notify maintainers

@Hodapp87 

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.55, NixOS, 23.05 (Stoat), 23.05.20231011.bd1cde4`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.5`
 - nixpkgs: `/nix/store/mbz4hixfgxq5b6vc0k3pp2iglcd4c353-source`

--- a/pkgs/development/compilers/osl/default.nix
+++ b/pkgs/development/compilers/osl/default.nix
@@ -24,13 +24,13 @@ let
 
 in stdenv.mkDerivation rec {
   pname = "openshadinglanguage";
-  version = "1.12.13.0";
+  version = "1.12.14.0";
 
   src = fetchFromGitHub {
     owner = "AcademySoftwareFoundation";
     repo = "OpenShadingLanguage";
     rev = "v${version}";
-    hash = "sha256-EVV7YHovWwbRju+uv8IK2wpcpoK1ndZ8yNRHzU8LUuE=";
+    hash = "sha256-x8t7uC4q29MkTCIS1rK0ICw78u5zZG+/zyhwttLOBr4=";
   };
 
   cmakeFlags = [

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -11756,6 +11756,9 @@ with pkgs;
 
   osl = libsForQt5.callPackage ../development/compilers/osl {
     boost = boost179;
+    libclang = llvmPackages_15.libclang;
+    clang = clang_15;
+    llvm = llvm_15;
   };
 
   osqp = callPackage ../development/libraries/science/math/osqp { };

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -11757,7 +11757,10 @@ with pkgs;
   osl = libsForQt5.callPackage ../development/compilers/osl {
     boost = boost179;
     libclang = llvmPackages_15.libclang;
-    clang = clang_15;
+    clang =
+      if stdenv.cc.libcxx != null
+      then (overrideLibcxx llvmPackages_15.stdenv).cc
+      else clang_15;
     llvm = llvm_15;
   };
 

